src/core/meshcop/tcat_agent.cpp

```diff
@@ -55,7 +56,10 @@ RegisterLogModule("TcatAgent");
 
 bool TcatAgent::VendorInfo::IsValid(void) const
 {
-    return mProvisioningUrl == nullptr || IsValidUtf8String(mProvisioningUrl) || mPskdString != nullptr;
+    return (mProvisioningUrl == nullptr ||
+            (IsValidUtf8String(mProvisioningUrl) &&
+             (static_cast<uint8_t>(strlen(mProvisioningUrl)) < kProvisioningUrlMaxLength))) ||
```

Comment:
Suggest using `StringLength()` instead. It requires specifying a maximum length to check, which adds a layer of safety (similar to `strnlen`). 

We generally avoid using `strlen()`  because of how it may access memory if the string is not properly null-terminated.

Reply:
fixed.